src/services/goToDefinition.ts

```diff
@@ -510,4 +522,21 @@ namespace ts.GoToDefinition {
                 return false;
         }
     }
+
+    function createDefinitionInfoFromNode(node: Node, name: string): DefinitionInfo {
+        const sourceFile = node.getSourceFile();
+        return {
+            fileName: sourceFile.fileName,
+            textSpan: createTextSpanFromNode(node, sourceFile),
+            kind: ScriptElementKind.label,
+            name,
+            containerKind: ScriptElementKind.unknown,
+            containerName: "",
+            contextSpan: createTextSpanFromNode(node, sourceFile),

```

Comment:
contextSpan needs to be undefined or switch keyword + expression ?

Reply:
@sheetalkamat why can't it be the node itself?